### PR TITLE
#42: CI Environment is not properly prepared such that CI fails

### LINKED ISSUE #42: CI Environment is not properly prepared such that CI fails
# Background 

- The CI sets the log output path for luigi
  - https://github.com/exasol/script-languages-container-ci/blob/e3da072536b6d0a786e2e1f3246a71ac735ae68e/exasol_script_languages_container_ci/lib/ci.py#L75
- However, it doesn't make sure that the directory exists
- This triggers the following bug in the itde https://github.com/exasol/integration-test-docker-environment/issues/359
- Fixing the bug in the itde might take a while, for that reason, we should fix it also here and improve the testing in that sense

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #42

### CODE REVIEW

File: test/unit_tests/test_ci_prepare.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+from exasol_integration_test_docker_environment.cli.options.system_options import DEFAULT_OUTPUT_DIRECTORY
+from exasol_integration_test_docker_environment.lib.base import luigi_log_config
+
+from exasol_script_languages_container_ci.lib.ci_prepare import CIPrepare
+
+
+@pytest.fixture(autouse=True)
+def mock_settings_env_vars():
+    with mock.patch.dict(os.environ, {}):
+        yield
```
Comment (Nicoretti) [inline test/unit_tests/test_ci_prepare.py:15]: is wiping the entire `env` always necessary? (autouse)

File: test/integration_tests/test_ci_build.py [hunk 2]
```diff
@@ -21,6 +25,10 @@ def test(input_docker_build_repository,
     test_type = "successful"
     flavor_path = str(flavors_path / test_type)
     test_container_folder = str(test_containers_folder / test_type)
+    logging.basicConfig(level=logging.DEBUG)
+    logging.info("test")
+    os.makedirs(".build_output/jobs", exist_ok=True)
```
Comment (tkilias) [inline test/integration_tests/test_ci_build.py:30]:
```suggestion
```

[the remark this answers is not in the document]
Comment (tkilias) [reply test/integration_tests/test_ci_build.py]: this was remaining from initial tests, I removed it

File: test/unit_tests/ci_calls.py [hunk 3]
```diff
@@ -1,11 +1,12 @@
 from unittest.mock import call
 
-from exasol_integration_test_docker_environment.lib.api.build_test_container import build_test_container
-from exasol_script_languages_container_tool.cli.commands import build, run_db_test, security_scan, push
-
 from test.unit_tests.test_env import test_env
 
 
+def prepare():
+    return call.prepare()
```
[the remark this answers is not in the document]
Comment (tkilias) [reply test/unit_tests/ci_calls.py:7]: only that the call definition in test_ci and test_release look similar

File: test/unit_tests/test_ci_prepare.py (hunk 1 above)
Comment (tkilias) [reply test/unit_tests/test_ci_prepare.py:15]: for these tests yes, the autouse is restricted to this module, if I understand pytest correct.

**Review by Nicoretti — APPROVED** [review]

**Pull request merged**